kv/dist_sender.go

```diff
@@ -511,31 +509,19 @@ func (ds *DistSender) Send(call client.Call) {
 					// re-run as part of a transaction for consistency. The
 					// case where we don't need to re-run is if the read
 					// consistency is not required.
-					if call.Args.Header().Txn == nil &&
-						args.Header().ReadConsistency != proto.INCONSISTENT {
+					if header.Txn == nil && header.ReadConsistency != proto.INCONSISTENT {
 						return util.RetryBreak, &proto.OpRequiresTxnError{}
 					}
 					// This next lookup is likely for free since we've read the
 					// previous descriptor and range lookups use cache
 					// prefetching.
 					descNext, err = ds.rangeCache.LookupRangeDescriptor(desc.EndKey)
-					// If this is the first step in a multi-range operation,
-					// additionally copy call.Args because we will have to
-					// mutate it as we talk to the involved ranges.
-					if len(responses) == 0 {
-						args = gogoproto.Clone(call.Args).(proto.Request)
-					}
 					// Truncate the request to our current range.
 					args.Header().EndKey = desc.EndKey
```

Comment:
`header` is a copy of `args.Header()`, not a pointer to it. I've changed this up though so that I only keep `endKey`. I've convinced myself that making a copy of `args.Header()` was unnecessary.